api/routers/access_control_changed.py

```diff
@@ -1,59 +0,0 @@
-import logging
-from typing import List
-
-from fastapi import APIRouter, Response, status
-from pydantic import BaseModel
-
-import api.cache as cache
-
-router = APIRouter()
-logger = logging.getLogger("micro-auth")
-
-
-class UserAccess(BaseModel):
-    id: str
-    access: str
-
-
-class ResourceAccess(BaseModel):
-    id: str
-    user_access: List[UserAccess]
-    public: bool
-    discoverable: bool
-    allow_private_sharing: bool
-    bucket_name: str
-
-
-class AccessControlChanged(BaseModel):
-    resources: List[ResourceAccess]
-
-
-@router.post("/hook/")
-async def set_auth(access_control_changed: AccessControlChanged, response: Response):
-    try:
-        for resource in access_control_changed.resources:
-            resource_id = resource.id
-            for user_access in resource.user_access:
-                user_id = user_access.id
-                access = user_access.access
-                cache.set_cache_xx(f"{user_id}:{resource_id}", access)
```

Comment:
what if this access_control_changed route is called for a user/resource pair that didn't previously exist? Will this cause issues with the redis -XX ?

Reply:
Yep, got it. But wouldn't we call this route/hook say on the creation of a new resource for a user that didn't have any previous resources or who had not previously used the s3 protocol?

Reply:
The hook/route would be called and in the case of a new user, the cache would not be set. When the user starts using the S3 protocol, we check the cache, get a miss and then hit the db, followed by filling the cache. If the access control changes for the user/resource (before the cache expires), then the cache will be updated by this route which is called by HydroShare.

Perhaps a diagram would help document how this behaves.

Reply:
sounds good, I wasn't sure if calling `set-xx` for non-existent key would raise an exception. No need for a diagram, your explanation aligns with my understanding, I just wasn't sure about the redis call in this case. But I see that it just returns a "nill reply"